unstructured/ingest/connector/weaviate.py

```diff
@@ -0,0 +1,166 @@
+import json
+import typing as t
+from dataclasses import dataclass
+
+from unstructured.ingest.error import DestinationConnectionError, SourceConnectionError
+from unstructured.ingest.interfaces import (
+    BaseConnectorConfig,
+    BaseDestinationConnector,
+    BaseIngestDoc,
+    WriteConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.utils import requires_dependencies
+
+
+@dataclass
+class SimpleWeaviateConfig(BaseConnectorConfig):
+    host_url: str
+    class_name: str
+    auth_keys: t.Optional[t.Dict[str, str]] = None
+
+
+@dataclass
+class WeaviateWriteConfig(WriteConfig):
+    batch_size: int = 100
+
+
+@dataclass
+class WeaviateDestinationConnector(BaseDestinationConnector):
+    write_config: WeaviateWriteConfig
+    connector_config: SimpleWeaviateConfig
+
+    @requires_dependencies(["weaviate"], extras="weaviate")
+    @DestinationConnectionError.wrap
+    def initialize(self):
+        from weaviate import Client
+
+        auth = self._resolve_auth_method()
+        self.client: Client = Client(url=self.connector_config.host_url, auth_client_secret=auth)
+
+    @requires_dependencies(["weaviate"], extras="weaviate")
+    def check_connection(self):
+        from weaviate import Client
+
+        try:
+            auth = self._resolve_auth_method()
+            _ = Client(url=self.connector_config.host_url, auth_client_secret=auth)
+        except Exception as e:
+            logger.error(f"Failed to validate connection {e}", exc_info=True)
+            raise SourceConnectionError(f"failed to validate connection: {e}")
+
+    def _resolve_auth_method(self):
+        if self.connector_config.auth_keys is None:
+            return None
+
+        if access_token := self.connector_config.auth_keys.get("access_token"):
+            from weaviate.auth import AuthBearerToken
+
+            return AuthBearerToken(
+                access_token=access_token,
+                refresh_token=self.connector_config.auth_keys.get("refresh_token"),
+            )
+        elif api_key := self.connector_config.auth_keys.get("api_key"):
+            from weaviate.auth import AuthApiKey
+
+            return AuthApiKey(api_key=api_key)
+        elif client_secret := self.connector_config.auth_keys.get("client_secret"):
+            from weaviate.auth import AuthClientCredentials
+
+            return AuthClientCredentials(
+                client_secret=client_secret, scope=self.connector_config.auth_keys.get("scope")
+            )
+        elif (username := self.connector_config.auth_keys.get("username")) and (
+            pwd := self.connector_config.auth_keys.get("password")
+        ):
+            from weaviate.auth import AuthClientPassword
+
+            return AuthClientPassword(
+                username=username, password=pwd, scope=self.connector_config.auth_keys.get("scope")
+            )
+        return None
+
+    def conform_dict(self, data: dict) -> None:
```

Comment:
Just curious, how similar is this to the `conform_dict` method in the azure cognitive search dest connector? Wondering if this should be extracted it so it can be more widely reused.

Reply:
So what was the conclusion from this. Is it similar enough that we can extract it to somewhere else and also create a unit test (since it is seems like the thing we should test... lots of little operations...) Thoughts @rbiseck3 ?

Reply:
Maybe we do it as a separate PR... This connector is almost ready to merge :)